hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/InodeTree.java

```diff
@@ -410,7 +410,16 @@ private boolean hasFallbackLink() {
     return rootFallbackLink != null;
   }
 
-  protected INodeLink<T> getRootFallbackLink() {
+  /**
+   * @return true if the root represented as internalDir. In LinkMergeSlash,
+   * there will be root to root mapping. So, root does not represent as
+   * internalDir.
+   */
+  public boolean isRootInternalDir() {
```

Comment:
Can you check which commit added this in trunk? Should that be backported before this commit?

Reply:
We found HDFS-15529 added this method (isRootInternalDir) with a few other changes to getChildFileSystem in ViewFileSystem. Unfortunately, new changes have been made since HDFS-15529 and made its pull request out-of-date (targetFileSystem in InodeLink is now private, instead of package private). Besides, file TestViewFileSystemOverloadSchemeWithHdfsScheme.java does not exist in branch-2.10 yet. So, we decide to leave out isRootInternalDir() for this PR. 

PR for HDFS-15529:
https://github.com/apache/hadoop/pull/2234